src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -206,6 +221,16 @@ public void delete(final String eventTypeName) throws EventTypeDeletionException
                 }
             }
         }
+        try {
+            if (category != null) {
```

Comment:
is there any specific reason, that you did not put the code inside this `if` within `try/catch` where event type object is available ?

Reply:
Yes, I wanted to publish KPI event only after the ET deletion is finished (including topic deletion). That's why I put it in a place after all topics are deleted.